requirements-dev.txt

```diff
@@ -13,6 +13,7 @@ testtools
 # Extra dependencies for tests/documentation code
 multimethod
 qiskit-ibm-runtime>=0.19
+pyarrow # avoid pandas warning
```

Comment:
I would just add `pyarrow` to the docs environments' `deps` in `tox.ini` instead for now to keep the test envs lighter but we might end up needing `pyarrow` in the future if Pandas does make it required.